sources/api/apiclient/src/main.rs

```diff
@@ -8,48 +17,98 @@ const DEFAULT_METHOD: &str = "GET";
 /// Stores user-supplied global arguments.
 #[derive(Debug)]
 struct Args {
-    verbosity: usize,
+    log_level: LevelFilter,
     socket_path: String,
 }
 
 impl Default for Args {
     fn default() -> Self {
         Self {
-            verbosity: 3,
+            log_level: LevelFilter::Info,
             socket_path: DEFAULT_API_SOCKET.to_string(),
         }
     }
 }
 
 /// Stores the usage mode specified by the user as a subcommand.
+#[derive(Debug)]
 enum Subcommand {
     Raw(RawArgs),
+    Reboot(RebootArgs),
+    Update(UpdateSubcommand),
 }
 
 /// Stores user-supplied arguments for the 'raw' subcommand.
+#[derive(Debug)]
 struct RawArgs {
     method: String,
     uri: String,
     data: Option<String>,
 }
 
+/// Stores user-supplied arguments for the 'reboot' subcommand.
+#[derive(Debug)]
+struct RebootArgs {}
+
+/// Stores the 'update' subcommand specified by the user.
+#[derive(Debug)]
+enum UpdateSubcommand {
+    Check(CheckArgs),
+    Apply(ApplyArgs),
+    Cancel(CancelArgs),
+}
+
+/// Stores user-supplied arguments for the 'update check' subcommand.
+#[derive(Debug)]
+struct CheckArgs {}
+
+/// Stores user-supplied arguments for the 'update apply' subcommand.
+#[derive(Debug)]
+struct ApplyArgs {
+    check: bool,
+    reboot: bool,
+}
+
+/// Stores user-supplied arguments for the 'update cancel' subcommand.
+#[derive(Debug)]
+struct CancelArgs {}
+
 /// Informs the user about proper usage of the program and exits.
 fn usage() -> ! {
     let msg = &format!(
         r"Usage: apiclient [SUBCOMMAND] [OPTION]...
 
         Global options:
             -s, --socket-path PATH     Override the server socket path.  Default: {socket}
-            -v, --verbose              Print extra information like HTTP status code.
+            --log-level                Desired amount of output; trace|debug|info|warn|error
+            -v, --verbose              Sets log level to 'debug'.  This prints extra info,
```

Comment:
Having both `--log-level` and `-v` is a tiny bit confusing, especially because I can't pass `-v -v ` or `-vvv` and get expected results.

I think I would bias towards just having `--log-level` unless you're trying to maintain some level of API compatibility with previous versions.

Reply:
There are definitely compromises for backward compatibility here, `-v` being one of them, and manual arg parsing being the other big one.  I didn't allow multiple `-v` levels because it didn't work before, and I'd like to encourage people toward `--log-level`, as you suggested.

Reply:
That's fair.  Carry on.